rskj-core/src/test/java/co/rsk/peg/PowpegMigrationTest.java

```diff
@@ -200,15 +204,16 @@ private void testChangePowpeg(
             argumentCaptor.capture()
         );
 
+        // TODO check this. maybe adding the FedType to the constructor solves it
```

Comment:
Is there anything to discuss here?

Reply:
oh yes. thank you. my concern is that we are checking
```
            case legacyErp:
                assertSame(ErpFederation.class, newPowPeg.getClass());
                break;
            case p2shErp:
                assertSame(ErpFederation.class, newPowPeg.getClass());
```

so no real difference between those two

And checking for the builders with instanceof approach is not that accurate